src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/CSharpConvertToRecordRefactoringProvider.cs

```diff
@@ -0,0 +1,791 @@
+            var propertiesToMove = originalDeclarationNode.Members
+                .Where(m => ShouldConvertProperty(m, originalType))
+                .Cast<PropertyDeclarationSyntax>()
+                .AsImmutable();
+
+            var modifiedClassTrivia = GetModifiedClassTrivia(propertiesToMove, originalDeclarationNode);
+
+            var propertiesToAddAsParams = propertiesToMove.SelectAsArray(p =>
+                SyntaxFactory.Parameter(
+                    GetModifiedAttributeListsForProperty(p),
+                    modifiers: default,
+                    p.Type,
+                    p.Identifier,
+                    @default: null));
+
+            // remove converted properties and reformat other methods
+            var membersToKeep = GetModifiedMembersForPositionalRecord(
+                originalType, originalDeclarationNode, semanticModel, propertiesToMove, cancellationToken);
+
+            // if we have a class, move trivia from class keyword to record keyword
+            // if struct, split trivia and leading goes to record, trailing goes to struct
+            var recordKeyword = SyntaxFactory.Token(SyntaxKind.RecordKeyword);
+            recordKeyword = originalType.TypeKind == TypeKind.Class
+                    ? recordKeyword.WithTriviaFrom(originalDeclarationNode.Keyword)
+                    : recordKeyword.WithLeadingTrivia(originalDeclarationNode.Keyword.LeadingTrivia);
+
+            var changedTypeDeclaration = SyntaxFactory.RecordDeclaration(
+                originalType.TypeKind == TypeKind.Class
+                    ? SyntaxKind.RecordDeclaration
+                    : SyntaxKind.RecordStructDeclaration,
+                originalDeclarationNode.AttributeLists,
+                originalDeclarationNode.Modifiers,
+                recordKeyword,
+                originalType.TypeKind == TypeKind.Class
+                    ? default
+                    : originalDeclarationNode.Keyword.WithTrailingTrivia(SyntaxFactory.ElasticMarker),
+                // remove trailing trivia from places where we would want to insert the parameter list before a line break
+                originalDeclarationNode.Identifier.WithTrailingTrivia(SyntaxFactory.ElasticMarker),
+                originalDeclarationNode.TypeParameterList?.WithTrailingTrivia(SyntaxFactory.ElasticMarker),
+                SyntaxFactory.ParameterList(SyntaxFactory.SeparatedList(propertiesToAddAsParams)),
+                // TODO: inheritance in the case where we inherit from a record (because we previously activated on one)
+                originalDeclarationNode.BaseList,
+                originalDeclarationNode.ConstraintClauses,
+                originalDeclarationNode.OpenBraceToken,
+                SyntaxFactory.List(membersToKeep),
+                originalDeclarationNode.CloseBraceToken,
+                semicolonToken: default)
+                .NormalizeWhitespace()
```

Comment:
calling NormalizeWhitespace should not be necessary.  It will also obliterate all user formatting within the record itself.  def should not do this.